### PR TITLE
Add document assigned message

### LINKED ISSUE #31: Trigger event when a user is granted permission to a document
Teedy already has the functionality to grant another user permission to a document, yet it is hard for the user to recognize any new documents that one is assigned to. I will trigger the `DOCUMENT_ASSIGEND` event in this case so that a message can be sent to the user.

### PR DESCRIPTION
**Pull request by kyminbb**
## Purpose

This PR resolves #31 so that a user can be notified with a message when permission to a document is granted.

## Changes

Teedy already has an endpoint for handling permission creation of a resource (`PUT /acl`). When the API is called and the type of the ACL target is `USER`, this PR triggers a `DOCUMENT_ASSIGNED` event which can be caught by `MessageAsyncListener` to produce a new message. It also tested the functionality by verifying the message generation in `AclResource`.

### CODE REVIEW

File: docs-web/src/main/webapp/package-lock.json [hunk 1]
```diff
@@ -1,8 +1,3071 @@
...
+      "integrity": "sha1-m7UwTS4LVmmLLHWLCKPqqdqlijk=",
+      "dev": true,
+      "engines": {
+        "node": ">=0.10.0"
+      }
+    }
+  },
   "dependencies": {
     "abbrev": {
       "version": "1.1.1",
```
Comment (LeoJung1) [inline docs-web/src/main/webapp/package-lock.json:3071]: Just a general question, but what is the purpose of this file?

File: docs-web/src/test/java/com/sismics/docs/rest/TestAclResource.java [hunk 2]
```diff
@@ -12,442 +13,450 @@
...
+                                                .param("color", "#ff0000")), JsonObject.class);
+
+                // acltag2 can edit document1
+                target().path("/document/" + document1Id).request()
+                                .cookie(TokenBasedSecurityFilter.COOKIE_NAME, acltag2Token)
+                                .post(Entity.form(new Form()
+                                                .param("title", "My super document 1")
+                                                .param("tags", tag1Id)
+                                                .param("language", "eng")), JsonObject.class);
+        }
```
Comment (LeoJung1) [inline docs-web/src/test/java/com/sismics/docs/rest/TestAclResource.java:461]: This file also?

**Review by LeoJung1 — APPROVED** [review]
Other files seem fine to me!

File: docs-web/src/test/java/com/sismics/docs/rest/TestAclResource.java (hunk 2 above)
Comment (kyminbb) [reply docs-web/src/test/java/com/sismics/docs/rest/TestAclResource.java:461]: This is a test for the `/acl` endpoint.

**Review by jihyoc — APPROVED** [review]
lgtm

File: docs-web/src/main/webapp/package-lock.json (hunk 1 above)
Comment (kyminbb) [reply docs-web/src/main/webapp/package-lock.json:3071]: It is for the frontend compilation purpose. It includes are the relevant configs (dependencies, versions, etc.)

**Pull request merged**